Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,25 +1,49 @@
-function setAlarm() {}
+var audio = new Audio("alarmsound.mp3"); // Sound file for the alarm
+let interval; // Variable to store the countdown timer
 
-// DO NOT EDIT BELOW HERE
+// Function to set the alarm
+function setAlarm() {
+    const input = document.getElementById("alarmSet").value; // Get user input
+    let timeRemaining = parseInt(input); // Convert input to an integer
 
-var audio = new Audio("alarmsound.mp3");
+    if (isNaN(timeRemaining) || timeRemaining <= 0) {
+        alert("Please enter a valid number!"); // Validate input
+        return;
+    }
 
-function setup() {
-  document.getElementById("set").addEventListener("click", () => {
-    setAlarm();
-  });
+    const timeDisplay = document.getElementById("timeRemaining"); // Display remaining time
+    timeDisplay.innerText = `Time Remaining: ${String(Math.floor(timeRemaining / 60)).padStart(2, "0")}:${String(timeRemaining % 60).padStart(2, "0")}`;
 
-  document.getElementById("stop").addEventListener("click", () => {
-    pauseAlarm();
-  });
+    interval = setInterval(() => {
+        if (timeRemaining <= 0) {
+            clearInterval(interval); // Stop the timer
+            playAlarm(); // Play the alarm sound
+            timeDisplay.innerText = "Time Remaining: 00:00";
```

Comment:
I don't think this is needed i.e. you don't need to set the time back to `Time Remaining: 00:00` . Your alarm still works as expected without it. The interval is already cleared so the time remaining doesn't change so it will just display `00:00`